src/pages/index.js

```diff
@@ -98,48 +149,91 @@ function fillProfileForm() {
 
 function openModal(modal) {
   modal.classList.add("modal_opened");
-  document.addEventListener("keydown", handleEscKey); // added
+  document.addEventListener("keydown", handleEscKey);
 }
 
 function closeModal(modal) {
   modal.classList.remove("modal_opened");
-  document.removeEventListener("keydown", handleEscKey); // added
+  document.removeEventListener("keydown", handleEscKey);
 }
-
+//complete
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
+
+  const submitBtn = evt.submitter;
+  // submitBtn.textContent = "Saving...";
+  setButtonText(submitBtn, true);
+
+  api
+  .editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value })
+  .then((data) => {
+  profileName.textContent = data.name; //using data argument instead of input values
+  profileDescription.textContent = data.about;
   closeModal(editModal);
+})
+.catch(console.error)
+.finally(() => {
+  setButtonText(submitBtn, false);
+});
 }
 
+// complete
+
 function handleAddCardFormSubmit(event) {
   event.preventDefault();
 
+  const submitBtn = event.submitter;
+  setButtonText(submitBtn, true);
+
   const inputValues = {
     name: cardNameInput.value,
     link: cardLinkInput.value,
   };
 
-  renderCard(inputValues, "prepend"); // Modified to use the new function
+  api.addCard(inputValues)
+  .then((newCard) => {
+    renderCard(newCard, "prepend");
+    closeModal(addModal);
+    cardForm.reset();
+    resetValidation(cardForm, settings);
```

Comment:
`    resetValidation(cardForm, settings);
`

`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submission. It’s a **waste of resources**. You need to call `disableButton` there.


```js

    disableButton(submitBtn, settings);

```